smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/api/request/service/RetrieveInstructorsConsultingHoursRequestServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package hu.unideb.smartcampus.service.api.request.service;
+
+import java.util.Calendar;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import hu.unideb.smartcampus.persistence.entity.ConsultingDateEntity;
+import hu.unideb.smartcampus.persistence.entity.FromToDateEmbeddedEntity;
+import hu.unideb.smartcampus.persistence.repository.InstructorRepository;
+import hu.unideb.smartcampus.service.api.MessageProcessingClass;
+import hu.unideb.smartcampus.service.api.domain.response.wrapper.InstructorConsultingHoursWrapper;
+import hu.unideb.smartcampus.service.api.domain.response.wrapper.inner.ConsultingHourWrapper;
+import hu.unideb.smartcampus.service.api.domain.response.wrapper.inner.FromToDateWrapper;
+import hu.unideb.smartcampus.shared.requestmessages.BaseRequestType;
+import hu.unideb.smartcampus.shared.requestmessages.RetrieveInstructorConsultingHours;
+import hu.unideb.smartcampus.shared.requestmessages.constants.RequestMessagesConstants;
+
+/**
+ * Retrieve instructor consulting dates service.
+ */
+@Service(RetrieveInstructorsConsultingHoursRequestServiceImpl.BEAN_NAME)
+@Transactional(propagation = Propagation.REQUIRED)
+public class RetrieveInstructorsConsultingHoursRequestServiceImpl
+    implements MessageProcessingClass<InstructorConsultingHoursWrapper> {
+
+  public static final String BEAN_NAME = "retrieveInstructorsConsultingHoursRequestServiceImpl";
+
+  @Autowired
+  private InstructorRepository instructorRepository;
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public InstructorConsultingHoursWrapper getResponse(Object object) {
+    RetrieveInstructorConsultingHours msg = (RetrieveInstructorConsultingHours) object;
```

Comment:
This method is called by:
hu.unideb.smartcampus.service.api.impl.MessageProcessingServiceImpl.getResult(BaseRequestType)

It has a null check before the method is called so I think null check is not required, besides you test it with null.